torch/distributions/binomial.py

```diff
@@ -109,3 +111,27 @@ def enumerate_support(self):
         values = values.view((-1,) + (1,) * len(self._batch_shape))
         values = values.expand((-1,) + self._batch_shape)
         return values
+
+    def _Elnchoosek(self):
+        # return expected value of log(nchoosek), log(n!),log(k!), log(n-k!)
+        # where k~Bin(n,p)
+        s = self.enumerate_support()
+        s[0] = 1  # 0! = 1
+        # x is factorial matrix i.e. x[k,...] = k!
+        x = torch.cumsum(s.log(), dim=0)
+        s[0] = 0
+        indices = [slice(None)] * x.dim()
+        indices[0] = torch.arange(x.size(0) - 1, -1, -1,
+                                  dtype=torch.long, device=x.device)
+        # x[tuple(indices)] is x reversed on first axis
+        lnchoosek = x[-1] - x - x[tuple(indices)]
```

Comment:
I think `x.flip(dim=0)` will exhibit same behaviour.

Reply:
weird, I tried using `flip` and it didn't work before- maybe I messed with arguments...